source/backend/api/Services/PropertyService.cs

```diff
@@ -209,9 +209,70 @@ public PropertyManagementModel GetPropertyManagement(long propertyId)
             propertyManagement.RelatedLeases = leaseCount;
             propertyManagement.LeaseExpiryDate = leaseExpiryDate.HasValue ? DateOnly.FromDateTime(leaseExpiryDate.Value) : null;
 
+            PropertyHasActiveLease(propertyLeases, out bool hasActiveLease, out bool hasActiveExpiryDate);
+            propertyManagement.HasActiveLease = hasActiveLease;
+            propertyManagement.ActiveLeaseHasExpiryDate = hasActiveExpiryDate;
+
             return propertyManagement;
         }
 
+        private static void PropertyHasActiveLease(IEnumerable<PimsPropertyLease> propertyLeases, out bool hasActiveLease, out bool hasActiveExpiryDate)
+        {
+            hasActiveLease = false;
+            hasActiveExpiryDate = false;
+
+            List<PimsLease> activeAgreementsList = propertyLeases.Select(x => x.Lease).ToList();
+            var activeLease = activeAgreementsList.FirstOrDefault(x => x.LeaseStatusTypeCode == LeaseStatusTypes.ACTIVE.ToString());
+
+            // No Active Lease
+            if (activeLease is not null)
+            {
+                // Lease is Active but has termination Date.
+                if (activeLease.TerminationDate.HasValue)
+                {
+                    return;
+                }
+                else
+                {
+                    var latestRemewal = activeLease.PimsLeaseRenewals.Where(x => x.IsExercised.HasValue && x.IsExercised.Value).OrderByDescending(x => x.CommencementDt).FirstOrDefault();
+                    if (latestRemewal is null) // No Renewal - Check only Lease dates.
+                    {
+                        if(activeLease.OrigExpiryDate.HasValue && activeLease.OrigExpiryDate.Value.Date >= DateTime.UtcNow.Date)
+                        {
+                            hasActiveLease = hasActiveExpiryDate = true;
+                        }
+                        else if(activeLease.OrigExpiryDate.HasValue && activeLease.OrigExpiryDate.Value.Date < DateTime.UtcNow.Date)
+                        {
+                            return;
+                        }
+                        else if (!activeLease.OrigExpiryDate.HasValue)
+                        {
+                            hasActiveLease = true;
+                        }
+                    }
+                    else
+                    {
+                        if(activeLease.OrigExpiryDate.HasValue && latestRemewal.ExpiryDt.HasValue)
+                        {
+                            hasActiveLease = hasActiveExpiryDate = activeLease.OrigExpiryDate.Value.Date >= DateTime.UtcNow.Date || latestRemewal.ExpiryDt.Value.Date >= DateTime.UtcNow.Date;
+                        }
+                        else if(activeLease.OrigExpiryDate.HasValue && !latestRemewal.ExpiryDt.HasValue)
+                        {
+                            hasActiveLease = true;
```

Comment:
wouldn't this also mean that the lease has an active expiry date?

Reply:
not quite sue, I am trying to cover for the scenarios not documented in the AC.